### ISSUE TITLE
ILXP Objects from JSON

### ISSUE THREAD

**Issue by sic2** [stacs-srg/storr#3]
Need to implement the method `LXP.readObjectIfPresent(reader, key)`

See: https://github.com/stacs-srg/storr/blob/master/src/main/java/uk/ac/standrews/cs/storr/impl/LXP.java#L442

**Comment by sic2** [comment]
Now working for cases such as:
```
{
    "glossary": {
        "title": "test glossary",
        "GlossDiv": {
            "title": "S",
            "valid": true
        }
    }
}
```


However, the LXP is still not formed correctly with the case where objects are contained inside arrays:

```
{
	"glossary": [{
		"title": "test title"
	}]
}
```

**Comment by sic2** [comment]
This commit fixes the problem above: e8dfe17c5b4f63fbb1b9245ebb9e91a52ef876d1

However, it seems that there are still problem with much more complex JSON structures.

Also, should there be a specific behaviour for cases where the JSON looks like this?

```
[
        {
		"title": "obj1"
	},
	{
		"title": "obj1"
	}
]
```
